include/aws/io/private/tls_channel_handler_shared.h

```diff
@@ -19,6 +19,12 @@ struct aws_tls_channel_handler_shared {
     struct aws_crt_statistics_tls stats;
 };
 
+struct aws_tls_delayed_shutdown_task {
```

Comment:
all the new tasks you added are simply calling `aws_channel_slot_on_handler_shutdown_complete()` which you could have just called synchronously

you could probably just revert this, and have a simpler diff overall